src/pages/matching/survey.tsx

```diff
@@ -9,23 +10,56 @@ import { CheckedOffIcon, CheckedOnIcon } from 'assets/icons';
 import { Seo } from 'components/common';
 import FeedbackTypeCheckbox from 'components/matching/FeedbackTypeCheckbox';
 import { PAGE_PATH } from 'constants/common';
+import { useAddToBlackList } from 'hooks/services/mutations/useBlockUser';
+import { useCreateMatchingFeedback } from 'hooks/services/mutations/useCreateMatchingFeedback';
+import { useRecentMatchingHistory } from 'hooks/services/queries/useRecentMatchingHistory';
 import { ScreenReaderOnly } from 'styles';
 
 const MatchingSurvey = () => {
   const router = useRouter();
 
-  const { control, register, handleSubmit } = useForm<MatchingFeedbackForm>();
+  const { register, handleSubmit } = useForm<MatchingFeedbackForm>();
 
-  const onSubmit: SubmitHandler<MatchingFeedbackForm> = async (data) => {
-    console.log(data); // FIXME: 실제 API 연동할 때 사용될 데이터 console.log 입니다.
+  const [isBlockUser, setIsBlockUser] = useState<boolean>(false);
 
-    await router.push(PAGE_PATH.main);
+  const { data: matchingHistory } = useRecentMatchingHistory();
+
+  const { mutate: createFeedbackMutate } = useCreateMatchingFeedback();
+
+  const { mutate: addToBlackListMutate } = useAddToBlackList();
+
+  const onRequestError = () => {
+    alert('의도하지 않은 에러가 발생하였습니다.');
+    void router.push(PAGE_PATH.main);
   };
 
-  const isBlockedMatching = useWatch({
-    control,
-    name: 'isBlockedMatching',
-  });
+  const onSubmit: SubmitHandler<MatchingFeedbackForm> = (formData) => {
+    if (!matchingHistory) return;
+
+    const { id, matchedUser } = matchingHistory;
+
+    if (isBlockUser) {
+      addToBlackListMutate(matchedUser.id, { onError: onRequestError });
+    }
+
+    createFeedbackMutate(
+      {
+        matchingHistoryId: id,
+        matchedUserId: matchedUser.id,
+        ...formData,
+      },
```

Comment:
p3;
matchingHistoryId, matchedUserId가 formData와 겹치는 값이 없을 수 있으나, formData에 덮어 씌워질 가능성이 있어 순서가 변경되는 것이 좋을 것 같아요!

```suggestion
      {
        ...formData,
        matchingHistoryId: id,
        matchedUserId: matchedUser.id,
      },
```

Reply:
이력 아이디와 매칭 유저는 formData에 포함되어 있지 않아 별생각 없이 위와 같이 구현을 했었습니다..!
제시해주신 방식으로 변경하였습니다

작업 커밋: c4e8f99